### PR TITLE
Perform the workgroup lookup using samba python bindings

### PR DESCRIPTION
**Pull request by dmulder**
This fixes [bug 1124390](https://bugzilla.opensuse.org/show_bug.cgi?id=1124390).

### CODE REVIEW

File: src/modules/SambaAPI.py [hunk 1]
```diff
@@ -2,20 +2,24 @@
...
-    from samba.net import Net
-    from samba.credentials import Credentials
-    from samba.dcerpc import nbt
-    net = Net(Credentials())
-    cldap_ret = net.finddc(domain=domain, flags=(nbt.NBT_SERVER_LDAP | nbt.NBT_SERVER_DS))
+    try:
+        net = Net(Credentials())
+        cldap_ret = net.finddc(domain=domain, flags=(nbt.NBT_SERVER_LDAP | nbt.NBT_SERVER_DS))
+    except NameError:
+        ycpbuiltins.y2error('The samba python bindings are not installed')
```
Comment (noelpower) [inline src/modules/SambaAPI.py]: Is the try/except necessary here ? I mean is it possible to throw an error if the package install fails forcing the import to fail thus making sure noone can call net_lookup, it's gonna be kindof ugly if you have to do a try/catch around every imported api statement

Comment (dmulder) [reply src/modules/SambaAPI.py]: So, you're suggesting I make the SambaAPI yast import fail somehow instead?

File: src/modules/SambaAPI.py [hunk 2]
```diff
@@ -2,6 +2,24 @@
 import_module('PackageSystem')
 import_module('Package')
 from yast import PackageSystem, Package
+if not PackageSystem.Installed('samba-python3'):
+    if Package.InstallAll(['samba-python3']):
+        from samba.net import Net
+        from samba.credentials import Credentials
+        from samba.dcerpc import nbt
+    else:
+        ycpbuiltins.y2error('Failed to import samba python bindings')
```
Comment (dmulder) [inline src/modules/SambaAPI.py]:
```suggestion
        ycpbuiltins.y2error('Failed to import samba python bindings')
        raise ImportError()
```

File: src/modules/SambaAPI.py (hunk 1 above)
Comment (noelpower) [reply src/modules/SambaAPI.py]: yep

File: src/modules/SambaAPI.py [hunk 3]
```diff
@@ -2,6 +2,22 @@
...
```
Comment (noelpower) [inline src/modules/SambaAPI.py]: maybe something like
``` diff
    else:
-        ycpbuiltins.y2error('Failed to import samba python bindings')
+        raise ImportError("Failed to install samba python bindings samba-python3")
```

That should work, should dump appropriate trace into y2log etc.
note: you will need https://github.com/yast/yast-python-bindings/pull/26
Then you won't need to check variiou

**Pull request merged**